compiler/packages/babel-plugin-react-compiler/src/HIR/HIR.ts

```diff
@@ -1933,7 +1946,8 @@ export function isStableTypeContainer(id: Identifier): boolean {
     isUseStateType(id) || // setState
     type_.shapeId === 'BuiltInUseActionState' || // setActionState
     isUseReducerType(id) || // dispatcher
-    type_.shapeId === 'BuiltInUseTransition' // startTransition
+    type_.shapeId === 'BuiltInUseTransition' || // startTransition
+    isUseOptimisticType(id) // setOptimistic
```

Comment:
tbh idk what this function is doing and why some hooks use type._shapeId directly?